ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -1137,7 +1137,7 @@ private String processTable(QB qb, ASTNode tabref) throws SemanticException {
 
     ASTNode tableTree = (ASTNode) (tabref.getChild(0));
 
-    String tabIdName = getUnescapedName(tableTree).toLowerCase();
+    String tabIdName = HiveUtils.getLowerCaseTableName(getUnescapedName(tableTree));
```

Comment:
what will happen if we simply remove toLowerCase() here?

Reply:
I am not very sure, that code is pretty old, so I preserved the old behaviour, it is used as a key in a map {{aliasToCTEs}}, not sure if it can break or lead to some perf penalties, if the query has same table with different cases or so....

Reply:
ok